components/views/MatesList/MatesList.tsx

```diff
@@ -1,15 +1,52 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { motion } from 'framer-motion'; // Import motion from framer-motion
-import useSWR from 'swr';
 import { fetcher } from '@/lib/fetcher';
 import { MateProps } from './Mate.interface';
 import { MateSkeleton } from './MateSkeleton';
 import { ErrorPage } from '@/components/Common/Error';
 import { Mate } from './Mate';
+import useSWRInfinite from 'swr/infinite';
 
 export const MatesList = () => {
-  const url = `/api/user/all`;
-  const { data, error, isLoading } = useSWR(url, fetcher);
+  const getKey = (pageIndex: number, previousPageData: MateProps[]) => {
+    let cursorId = '';
+    if (previousPageData) {
+      const lastMate = previousPageData[previousPageData.length - 1];
+      if (!lastMate) return null;
+      cursorId = lastMate.id;
+    }
+
+    if (pageIndex === 0) return `/api/user/all`;
+
+    return `/api/user/mates?cursorId=${cursorId}&limit=15`;
```

Comment:
/api/user/mates?cursorId=${cursorId}&limit=15`;

may i know the reason you created a seperate API? i believe this can be done within single api `api/user/all`.

Reply:
please check projects page, how it's done with single api only. lmk if you have any queries.

Reply:
@rohitdasu  i have initially done this in the same api, but in the feature description you mentioned that instead of modifying existing api, create a separate api, thats why...

Reply:
@csagar131 my bad, could you check if the api/user/all is used anywhere else? if it's used anywhere else then keep it as it is. if it's not used anywhere else then you can modify the existing api.

Reply:
Hi @rohitdasu  i have made the changes in the existing(all) api only as this was not used anywhere else in the project.

Reply:
Great